The great renaming

sverhoeven opened the pull request:
Redo of #56, but now for reals.

Endpoints:
```python
@app.get("/borders")
@app.get("/defaults/pathway/global")
@app.get("/map/<year>/<effortSharing>")
@app.get("/options/pathway/global")
@app.get("/regions")
@app.get("/regions/<region>")
@app.get("/statistics/budget/global")
@app.get("/statistics/gap/global")
@app.get("/statistics/ndc/projections/<region>")
@app.get("/statistics/ndc/reductions/<region>")
@app.get("/statistics/reductions/<region>")
@app.get("/timeseries/<region>/emissions/allocations")
@app.get("/timeseries/<region>/emissions/historical")
@app.get("/timeseries/<region>/policies/<policy>")
@app.get("/timeseries/global/emissions/pathway")
```

Other renames:
- "principle" and "effort sharing" to "allocation method"
- iso to region
- choices 2 options

This PR also 
- removes some unused code.
- adds ruff linter formatter for ws.py file
- centralized defaults of sliders to ws.py route, fixes #57

sverhoeven committed:
Rename pathwayCarbon to globalPathway

sverhoeven committed:
Apply suggestions from https://github.com/pbl-nl/website-carbon-budget-explorer/pull/56/files

Now have following routes:
/borders
/map/<year>/<effortSharing>
/options/pathway/global
/regions
/regions/<region>
/statistics/budget/global
/statistics/gap/global
/statistics/ncdProjections/<region>
/statistics/ndcReductions/<region>
/statistics/reductions/<region>
/timeseries/<region>/emissions/historical
/timeseries/<region>/policies/<policy>
/timeseries/<region>/principles
/timeseries/global/emissions/pathway

sverhoeven committed:
Rename iso to region

sverhoeven committed:
Drop sentry in sveltekit

It was never added to sveltket client/server code. We use the sentry in ws.py to capture errors.

sverhoeven committed:
typing hard

sverhoeven committed:
lowercase routes

sverhoeven committed:
Remove unused gap from /global

sverhoeven committed:
Too eagerly remove gap, it is still used elsewhere

sverhoeven committed:
Remove unused gap info

sverhoeven committed:
Make sure /statistics/ndc/reductions/<region> can handle null

sverhoeven committed:
Dont use global nesting on global page + formatting

sverhoeven committed:
Rename choices 2 options

To match ws.py:globalPathwayOptions endpoint

Peter9192 commented:
> ```python
> @app.get("/timeseries/global/emissions/pathway")
> ```

shouldn't there also be a `region` pathway for this?

Peter9192 commented on ws.py:
```diff
@@ -116,15 +116,14 @@ def pathwayChoices():
-def pathwaySelection():
-    choices = pathwayChoices()
+def globalPathwayChoices():
+    # TODO fix slider settings
```
should this be fixed here or in follow up?

sverhoeven replied:
done in 5ca686fb416c2509f377a111ef5ab39d360e2135

Peter9192 reviewed (commented):
Code changes look good, still need to test.

sverhoeven commented:
> > ```python
> > @app.get("/timeseries/global/emissions/pathway")
> > ```
> 
> shouldn't there also be a `region` pathway for this?

The region pages uses /timeseries/<region>/principles to show line from 2021-2100 per effort sharing / principle / allocation method

sverhoeven committed:
In ws.py rename principle/effortsharing 2 allocation method + Added ruff

Renamed routes:
- /map/<year>/<effortSharing> 2 /map/<year>/<allocation_method>
- /timeseries/<region>/principles 2 /timeseries/<region>/emissions/allocations

sverhoeven committed:
Make api.ts sync with ws.py

sverhoeven committed:
Started renaming "principle" and "effort sharing" to "allocation method"

sverhoeven commented:
@MarkMDekker On user facing side we are using "allocation method" or "method of allocation", however in code we are using "effort sharing" and "principle". I would like to rename those to "allocation method".  Is that OK? 

Started on it in #72, need to know if I should continue.

MarkMDekker commented:
allocation method is actually better. So yes, please continue doing that!

sverhoeven committed:
Format

sverhoeven committed:
Renaming "principle" and "effort sharing" to "allocation method"

sverhoeven committed:
Also rename on home page

sverhoeven committed:
Centralize defaults to ws.py:global_pathway_defaults method

Peter9192 commented:
Are the endpoints from the top post documented somewhere? Would be nice to be able to easily consult this again later.

sverhoeven committed:
Document how to list routes of ws.py

sverhoeven commented:
> Are the endpoints from the top post documented somewhere? Would be nice to be able to easily consult this again later.

Added command to README
```
flask --app ws:app routes -s rule
Endpoint                 Methods  Rule                                      
-----------------------  -------  ------------------------------------------
borders                  GET      /borders                                  
global_pathway_defaults  GET      /defaults/pathway/global                  
allocation_map           GET      /map/<year>/<allocation_method>           
global_pathway_options   GET      /options/pathway/global                   
regions                  GET      /regions                                  
region                   GET      /regions/<region>                         
static                   GET      /static/<path:filename>                   
budget                   GET      /statistics/budget/global                 
gap                      GET      /statistics/gap/global                    
ndc_projections          GET      /statistics/ndc/projections/<region>      
ndc_reductions           GET      /statistics/ndc/reductions/<region>       
allocation_reduction     GET      /statistics/reductions/<region>           
emission_allocations     GET      /timeseries/<region>/emissions/allocations
historical_emissions     GET      /timeseries/<region>/emissions/historical 
policy                   GET      /timeseries/<region>/policies/<policy>    
global_emission_pathway  GET      /timeseries/global/emissions/pathway  
```
(Did not list routes in readme to prevent out of sync problems)
